### PR TITLE
RAMPS: define FAN1 pin for printer+cutten configuration

### PR DESCRIPTION
**Pull request by hwmland**
### Description

RAMPS users running a combined Cutter and 3D printer when SPINDLE or LASER _FEATURE is defined should have part fan pin defined.

### Requirements

RAMPS based pin config including boards like the MKS_GEN_L etc.
3D printer + cutter combo

### Benefits

User can run the combination of features as desired. FAN1_PIN is defined again on mosfet D

### Configurations

#define BOARD_MKS_GEN_L
#define LASER_FEATURE
#define TEMP_SENSOR_BED 1

### Related Issues

#23406

### CODE REVIEW

**Comment by thinkyhead** [comment]
The part fan is typically `FAN_PIN`.

Previously the pins were assigned based on `FET_ORDER_SF` but that is no longer set for the spindle / laser, bringing along this first evident side-effect.

In order to return to the previously-established behavior, you might try applying `HAS_CUTTER` as an additional condition alongside `FET_ORDER_SF` in the RAMPS pins file.

**Comment by thinkyhead** [comment]
Please check if the updated FET layout conditions produce a workable result. Presumably, the extruder will be on E0, the cutter will be on E1, and the bed or fan will be on the next available PWM port. The layout can be forced by defining `FET_ORDER_xxx` to your preference, if the automatic choice doesn't work for your setup.

File: Marlin/src/pins/pins.h [hunk 1]
```diff
@@ -37,24 +37,30 @@
...
+  #elif MB(RAMPS_13_EEB, RAMPS_14_EEB, RAMPS_PLUS_EEB, RAMPS_14_RE_ARM_EEB, RAMPS_SMART_EEB, RAMPS_DUO_EEB, RAMPS4DUE_EEB)
+    #define FET_ORDER_EEB 1
+  #elif MB(RAMPS_13_EFF, RAMPS_14_EFF, RAMPS_PLUS_EFF, RAMPS_14_RE_ARM_EFF, RAMPS_SMART_EFF, RAMPS_DUO_EFF, RAMPS4DUE_EFF)
+    #define FET_ORDER_EFF 1
+  #elif MB(RAMPS_13_EEF, RAMPS_14_EEF, RAMPS_PLUS_EEF, RAMPS_14_RE_ARM_EEF, RAMPS_SMART_EEF, RAMPS_DUO_EEF, RAMPS4DUE_EEF)
+    #define FET_ORDER_EEF 1
+  #elif MB(RAMPS_13_SF,  RAMPS_14_SF,  RAMPS_PLUS_SF,  RAMPS_14_RE_ARM_SF,  RAMPS_SMART_SF,  RAMPS_DUO_SF,  RAMPS4DUE_SF)
+    #define FET_ORDER_SF 1
+  #elif TEMP_SENSOR_BED
+    #define FET_ORDER_EEB 1
```
Comment (hwmland) [inline Marlin/src/pins/pins.h]: Are you sure it should not be EFB?

File: Marlin/src/pins/pins.h [hunk 2]
```diff
@@ -37,24 +37,30 @@
...
+  #elif HAS_MULTI_HOTEND || (HAS_EXTRUDERS && HAS_CUTTER)
+    #if TEMP_SENSOR_BED
```
Comment (hwmland) [inline Marlin/src/pins/pins.h:52]: This condition can never be true (because of new line 51)

File: Marlin/src/pins/pins.h [hunk 3]
```diff
@@ -37,24 +37,30 @@
...
+      #define FET_ORDER_EEB 1
+    #else
+      #define FET_ORDER_EEF 1
+    #endif
+  #elif TEMP_SENSOR_BED
```
Comment (hwmland) [inline Marlin/src/pins/pins.h:57]: duplicit to line 51

**Comment by hwmland** [comment]
> The part fan is typically `FAN_PIN`.
> 

My fault, so sorry. I did not check properly wiring and I thought that FAN=>extruder fan, FAN1=>part fan. Reality is as you say.

**Comment by thinkyhead** [comment]
I expect this still improves the automatic FET assignment, and I had been meaning to allow users to override the `FET_ORDER_*` options anyway.

**Comment by hwmland** [comment]
> I expect this still improves the automatic FET assignment, and I had been meaning to allow users to override the `FET_ORDER_*` options anyway.

Having possibility to define ```FET_ORDER_``` explicitly solves problem I think. Automatic configuration does not work for me now, but it's OK for combined (3D+cutter) configurations. I'd perhaps even add warning when such configuration is selected and ```FET_ORDER_``` is not defined explicitely.
Anyway - OK for me, let's close my original issue once this get merged. Thank you!

**Pull request merged**